OneSignalSDK/onesignal/core/src/main/java/com/onesignal/core/internal/operations/impl/OperationRepo.kt

```diff
@@ -334,7 +339,14 @@ internal class OperationRepo(
         val delayFor = max(delayForOnRetries, retryAfterSecondsNonNull * 1_000)
         if (delayFor < 1) return
         Logging.error("Operations being delay for: $delayFor ms")
-        delay(delayFor)
+        var wakeMessage = LoopWaiterMessage(false, 0)
+        while (!wakeMessage.force) {
+            val waitedTheFullTime =
+                    withTimeoutOrNull(delayFor) {
+                        wakeMessage = waiter.waitForWake()
```

Comment:
Thinking about this again it would be better to create a different `waiter` for this. (calling it something like `retryWaiter` maybe?). After seeing the code again I have concerns how things will interact on the same `waiter`, as I don't know thinking `wake()` will wake this one and other existing. Sorry for the orignal suggestion of using the same `waiter` could work.

So here I think we can simply this logic, remove `waitedTheFullTime` as we want to break out of this wait right away if consent becomes true. Then we can call `wake()` on orignal waiter, without force or a time so it counts it as a "normal" operation, so we can batch other things like tags together.